google/cloud/opentelemetry/internal/monitored_resource.cc

```diff
@@ -52,7 +52,7 @@ struct AsStringVisitor {
 
 struct OTelKeyMatch {
   std::vector<std::string> otel_keys;
-  absl::optional<std::string> fallback = absl::nullopt;
+  std::string fallback;
```

Comment:
might need to explicitly say `std::string fallback = {}` for our initializer lists to work

Reply:
I think this works already – `std::string` default constructs to empty string and the compiler doesn’t complain?

Reply:
So I am not smart enough to tell you why the compiler is complaining. (and only in cmake-based builds :shrug:). But it is complaining:

![image](https://github.com/user-attachments/assets/30406f9f-0dc2-4018-983b-0e70fb3382f4)

e.g. this build: https://github.com/googleapis/google-cloud-cpp/pull/14923/checks?check_run_id=35680003918

Reply:
I think the simplest build to test on is:

```
ci/cloudbuild/build.sh -t demo-fedora-pr
```

Reply:
It looks like we had to do the same thing (explicitly initialize the field to its default value) when the type was `absl::optional<std::string>`. :shrug:

Reply:
Ah you’re right. Fixed now. Thanks for the build command.

Reply:
Haha, `clang-tidy` doesn’t like this:

```
error: initializer for member 'fallback' is redundant [readability-redundant-member-init,-warnings-as-errors]
Step #2:    55 |   std::string fallback = {};
```

I guess I’ll have to find the syntax to make it ignore this warning? Probably a better solution than ignoring the above `-Wmissing-field-initializers` warnings

Reply:
Uuuuuuuggghh...... I just added a commit to silence the compiler warning. I will deal with it later.[^1]

[^1]: if I feel like it.